service/src/main/scala/ch/epfl/bluebrain/nexus/storage/client/StorageClient.scala

```diff
@@ -0,0 +1,264 @@
+package ch.epfl.bluebrain.nexus.storage.client
+
+import akka.NotUsed
+import akka.actor.ActorSystem
+import akka.http.scaladsl.client.RequestBuilding._
+import akka.http.scaladsl.model.Multipart.FormData
+import akka.http.scaladsl.model.Multipart.FormData.BodyPart
+import akka.http.scaladsl.model.Uri.Path.{Segment, Slash}
+import akka.http.scaladsl.model._
+import akka.http.scaladsl.model.headers.OAuth2BearerToken
+import akka.http.scaladsl.unmarshalling.{FromEntityUnmarshaller, Unmarshaller}
+import akka.stream.scaladsl.Source
+import akka.util.ByteString
+import cats.effect.{Effect, IO, LiftIO}
+import cats.implicits._
+import cats.effect.implicits._
+import ch.epfl.bluebrain.nexus.commons.http.HttpClient
+import ch.epfl.bluebrain.nexus.commons.http.HttpClient._
+import ch.epfl.bluebrain.nexus.commons.http.JsonLdCirceSupport._
+import ch.epfl.bluebrain.nexus.rdf.implicits._
+import ch.epfl.bluebrain.nexus.iam.client.IamClientError.{Forbidden, Unauthorized}
+import ch.epfl.bluebrain.nexus.iam.client.types._
+import ch.epfl.bluebrain.nexus.storage.client.StorageClient._
+import ch.epfl.bluebrain.nexus.storage.client.StorageClientError._
+import ch.epfl.bluebrain.nexus.storage.client.config.StorageClientConfig
+import ch.epfl.bluebrain.nexus.storage.client.types.{FileAttributes, LinkFile, ServiceDescription}
+import com.typesafe.scalalogging.Logger
+import io.circe
+import io.circe.parser.parse
+import io.circe.{DecodingFailure, ParsingFailure}
+
+import scala.concurrent.{ExecutionContext, Future}
+import scala.reflect.ClassTag
+import scala.util.control.NonFatal
+
+class StorageClient[F[_]] private[client] (
+    config: StorageClientConfig,
+    attributes: HttpClient[F, FileAttributes],
+    source: HttpClient[F, AkkaSource],
+    serviceDesc: HttpClient[F, ServiceDescription],
+    emptyBody: HttpClient[F, NotUsed]
+)(implicit F: Effect[F], ec: ExecutionContext) {
+
+  /**
+    * Fetches the service description information (name and version)
+    */
+  def serviceDescription: F[ServiceDescription] =
+    serviceDesc(Get(config.iri.asAkka))
+
+  /**
+    * Checks that the provided storage bucket exists and it is readable/writable.
+    *
+    * @param name the storage bucket name
+    */
+  def exists(name: String)(implicit cred: Option[AuthToken]): F[Boolean] = {
```

Comment:
Use `AccessToken` instead of `AuthToken`. AuthToken is something coming from the AimClient that we are not gonna use anymore.